styx-scheduler-service/src/main/java/com/spotify/styx/TriggerManager.java

```diff
@@ -71,30 +72,30 @@ void tick() {
       }
     } catch (IOException e) {
       LOG.warn("Couldn't fetch global enabled status, skipping this run.");
-      return ;
+      return;
     }
 
-    final Map<Workflow, TriggerInstantSpec> map;
-    final Set<WorkflowId> enabled;
+    final Map<Workflow, TriggerInstantSpec> canBeTriggeredWorkflows;
+    final Set<WorkflowId> enabledWorkflows;
     try {
-      map = storage.workflowsWithNextNaturalTrigger();
-      enabled = storage.enabled();
+      canBeTriggeredWorkflows = storage.workflowsWithNextNaturalTrigger();
+      enabledWorkflows = storage.enabled();
     } catch (IOException e) {
       LOG.warn("Couldn't fetch workflows to trigger, skipping this run.");
       return;
     }
 
     final Instant now = time.get();
-    map.entrySet().forEach(entry -> {
-      final Workflow workflow = entry.getKey();
-      final Schedule schedule = workflow.configuration().schedule();
-      final TriggerInstantSpec instantSpec = entry.getValue();
-
-      if (now.isBefore(instantSpec.offsetInstant())) {
-        return;
-      }
+    canBeTriggeredWorkflows.entrySet().stream()
+        .filter(entry -> !now.isBefore(entry.getValue().offsetInstant()))
```

Comment:
It was amazing we have test cases replying on exact time matching. Good catch of that!